ui/ts/components/navbar.ts

```diff
@@ -29,18 +29,22 @@ module Components {
       isActive: (t: Target) => boolean;
     }
 
-    export function controller(ts: TargetSet): any {}
+    export function controller(): any {}
 
-    export function view(ctrl: any, ts: TargetSet): _mithril.MithrilVirtualElement {
-      return m("ul.navigation-sidebar", _.map(ts.targets(), (t: Target) =>
+    export enum NavbarOrientation {Vertical, Horizontal};
+
+    export function view(ctrl: any, args: {ts: TargetSet; orientation: NavbarOrientation; }): _mithril.MithrilVirtualElement {
+      return m(
+        (args.orientation === NavbarOrientation.Vertical ? "ul.navigation-sidebar" : "ul.navigation"),
```

Comment:
And again, I don't see this as being explicit, I see this as needlessly pushing design information in the code.  There are definitely times that the code has to be aware of the design; this is not one of them. Unless the navigation bar was going to *change* orientation at some point, *and* that orientation change required information to be attached to nodes at this level, there's really no reason for this component to know about it.  This component basically exists only to maintain an "active" link, which is something that cannot be tracked correctly at the CSS level.